kernel/comps/virtio/src/device/input/device.rs

```diff
@@ -244,23 +242,19 @@ impl EventTable {
         let vm_segment = VmAllocOptions::new(1).alloc_contiguous().unwrap();
 
         let default_event = VirtioInputEvent::default();
-        let iter = iter::repeat(&default_event);
+        let iter = iter::repeat(&default_event).take(EVENT_SIZE);
         let nr_written = vm_segment.write_vals(0, iter, 0).unwrap();
         assert_eq!(nr_written, EVENT_SIZE);
 
         let stream = DmaStream::map(vm_segment, DmaDirection::FromDevice, false).unwrap();
         Self { stream, num_events }
     }
 
-    fn get(&self, idx: usize) -> EventBuf<'_> {
+    fn get(&self, idx: usize) -> EventBuf {
         assert!(idx < self.num_events);
 
         let offset = idx * EVENT_SIZE;
-        EventBuf {
-            event_table: self,
-            offset,
-            size: EVENT_SIZE,
-        }
+        SafePtr::new(self.stream.clone(), offset)
```

Comment:
I think it should be ok to construct a safe pointer by borrowing the DMA stream, which is cheaper than cloning.

Reply:
I have Inherited `HasDaddr` for `&T` if `T` implements `HasDaddr`. So currently `EventBuf` becomes `SafePtr<VirtioInputEvent, &'a DmaStream>`, the `clone` is avoided.